Flaky test: GroupByQueryRunnerFailureTest.testTimeoutExceptionOnQueryable

The unit test `GroupByQueryRunnerFailureTest.testTimeoutExceptionOnQueryable` failed in this build: "[Build and test on ARM64 CPU architecture (1)](https://app.travis-ci.com/github/apache/druid/jobs/573989699)". The PR in question added new code unrelated to native queries, so this is likely a flaky test.

```text
[INFO] Running org.apache.druid.query.groupby.GroupByQueryRunnerFailureTest
[ERROR] Tests run: 20, Failures: 3, Errors: 0, Skipped: 0, Time elapsed: 4.855 s <<< FAILURE! - in org.apache.druid.query.groupby.GroupByQueryRunnerFailureTest
[ERROR] org.apache.druid.query.groupby.GroupByQueryRunnerFailureTest.testTimeoutExceptionOnQueryable[noRollupRtIndex]  Time elapsed: 0.016 s  <<< FAILURE!
java.lang.AssertionError: Expected test to throw an instance of org.apache.druid.query.QueryTimeoutException
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.rules.ExpectedException.failDueToMissingException(ExpectedException.java:278)
	at org.junit.rules.ExpectedException.access$100(ExpectedException.java:111)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:264)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	...
```

--- a/integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java
+++ b/integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java
@@ -53,6 +53,7 @@ import org.testng.annotations.Test;
 import javax.net.ssl.SSLContext;
 import javax.net.ssl.SSLException;
 import javax.ws.rs.core.MediaType;
+
 import java.io.IOException;
 import java.net.URL;
 
@@ -445,8 +446,11 @@ public class ITTLSTest
       catch (RuntimeException re) {
         Throwable rootCause = Throwables.getRootCause(re);
 
-        if (rootCause instanceof IOException && ("Broken pipe".equals(rootCause.getMessage())
-                                                 || "Connection reset by peer".contains(rootCause.getMessage()))) {
+        if (rootCause instanceof IOException
+            && (null != rootCause.getMessage())
+            && ("Broken pipe".equals(rootCause.getMessage())
+            || "Connection reset by peer".contains(rootCause.getMessage()))
+        ) {
           if (retries > MAX_CONNECTION_EXCEPTION_RETRIES) {
             Assert.fail(StringUtils.format(
                 "Broken pipe / connection reset retries exhausted, test failed, did not get %s.",

--- a/processing/src/test/java/org/apache/druid/query/groupby/GroupByQueryRunnerFailureTest.java
+++ b/processing/src/test/java/org/apache/druid/query/groupby/GroupByQueryRunnerFailureTest.java
@@ -29,6 +29,7 @@ import org.apache.druid.collections.CloseableStupidPool;
 import org.apache.druid.collections.ReferenceCountingResourceHolder;
 import org.apache.druid.java.util.common.concurrent.Execs;
 import org.apache.druid.java.util.common.granularity.Granularities;
+import org.apache.druid.java.util.common.guava.Sequences;
 import org.apache.druid.query.DruidProcessingConfig;
 import org.apache.druid.query.QueryCapacityExceededException;
 import org.apache.druid.query.QueryContexts;
@@ -296,7 +297,17 @@ public class GroupByQueryRunnerFailureTest
           }
         }
     );
-    QueryRunner<ResultRow> mergeRunners = factory.mergeRunners(Execs.directExecutor(), ImmutableList.of(runner));
+    QueryRunner<ResultRow> mockRunner = (queryPlus, responseContext) -> {
+      try {
+        Thread.sleep(100);
+      }
+      catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+      return Sequences.empty();
+    };
+
+    QueryRunner<ResultRow> mergeRunners = factory.mergeRunners(Execs.directExecutor(), ImmutableList.of(runner, mockRunner));
     GroupByQueryRunnerTestHelper.runQuery(factory, mergeRunners, query);
   }
 }

